rb/lib/selenium/webdriver/bidi/log_handler.rb

```diff
@@ -21,8 +21,9 @@ module Selenium
   module WebDriver
     class BiDi
       class LogHandler
-        ConsoleLogEntry = BiDi::Struct.new(:level, :text, :timestamp, :stack_trace, :type, :source, :method, :args)
-        JavaScriptLogEntry = BiDi::Struct.new(:level, :text, :timestamp, :stack_trace, :type, :source)
+        BASE_LOG_ENTRY = %i[level source text timestamp stacktrace].freeze
+        ConsoleLogEntry = WebDriver::Types::Struct.define(*BASE_LOG_ENTRY, :method, :args, :type)
+        JavaScriptLogEntry = WebDriver::Types::Struct.define(*BASE_LOG_ENTRY, :type)
```

Comment:
So @titusfortner I have one question here:

Since we want to use Data for immutable objects, shouldn't the console log event we get from the browser be immutable? 

But I really like the whole idea on the PR